restful_api/tests/test_self_enroll.py

```diff
@@ -0,0 +1,39 @@
+
+from flask import url_for
+
+def test_get_self_enrollment_for_one_engineer(client, db, self_enroll_factory,   admin_headers):
+    # test 404 
+    self_enrollment_url = url_for('api.self_enroll_by_ids', eng_id="9999", course_id="9999")
+    rep = client.get(self_enrollment_url, headers=admin_headers)
+    assert rep.status_code == 404
+
+    self_enrollments = self_enroll_factory.create_batch(5)  
+    db.session.add_all(self_enrollments)
+    db.session.commit()
+
+    # test get self enrollment
+    self_enrollment_url = url_for('api.self_enroll_by_ids', eng_id=self_enrollments[0].eng_id, course_id=self_enrollments[0].course_id)
+    rep = client.get(self_enrollment_url, headers=admin_headers)
+    assert rep.status_code == 200
+    assert rep.get_json()["enrollment"]["eng_id"] == self_enrollments[0].eng_id
```

Comment:
Can add error lines such as -

```
assert rep.get_json()["enrollment"]["eng_id"] == self_enrollments[0].eng_id, "Incorrect eng_id in self enrolment"
```

Sry i may have missed in parts of my code too.